flakey e2e test - topic/no_blocking.test.w

Error:

```
FAIL  src/test_corpus/sdk_tests/topic/no_blocking.test.w.test.ts > wing test -t sim
 Test Files  1 failed | 117 passed | 1 skipped (119)
AssertionError: fail ┌ no_blocking.test.wsim » root/env0/test:topic subscribers are invoked without blocking
     │ Error: assertion failed: counter.peek() == 1
     │     at C:\Users\RUNNER~1\AppData\Local\Temp\wing-bundles-7EWYwk\index.js:78:23
     │     at $Closure2.handle (C:\Users\RUNNER~1\AppData\Local\Temp\wing-bundles-7EWYwk\index.js:79:15)
     │     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
     └     at async exports.handler (C:\Users\RUNNER~1\AppData\Local\Temp\wing-bundles-7EWYwk\index.js:21568:10)
```

Build log: https://github.com/winglang/wing/actions/runs/6813446280/job/18528133970?pr=4855

--- a/examples/tests/sdk_tests/topic/no_blocking.test.w
+++ b/examples/tests/sdk_tests/topic/no_blocking.test.w
@@ -18,8 +18,8 @@ test "topic subscribers are invoked without blocking" {
   // since the subscriber sleeps for 3 seconds, the counter should still be 0
   assert(counter.peek() == 0);
 
-  util.sleep(6s);
-
   // after a while, the counter should be 1
+  util.waitUntil(inflight () => { return counter.peek() == 1; }, interval: 1s, timeout: 30s);
+
   assert(counter.peek() == 1);
 }

